src/components/forms/SoftRequiredErrors/utils.ts

```diff
@@ -0,0 +1,230 @@
+import type {
+  AnyComponentSchema,
+  Column,
+  ColumnsComponentSchema,
+  EditGridComponentSchema,
+  FieldsetComponentSchema,
+} from '@open-formulieren/types';
+import {getIn} from 'formik';
+
+import type {GetRegistryEntry} from '@/registry/types';
+
+export interface MissingFields {
+  /**
+   * The path to the component in the form values dict, used to link the "missing field"
+   * entry to the corresponding form component.
+   *
+   * Each key, from the root to the component, is joined together in a period seperated
+   * string. For children of `editgrid` and `columns` components, the children index is
+   * included in the path.
+   *
+   * Example:
+   * Lets assume you have a columns component with the key "cars", which has a
+   * soft-required textfield in its first column with the key "brand".
+   *
+   * In this situation, the `pathToComponent` will be: "cars.0.brand".
+   */
+  pathToComponent: string;
+  /**
+   * The label to show for the "missing field" entry. This is typically the label of the
+   * empty soft-required component.
+   *
+   * When the missing field is located inside a `editgrid` or `fieldset` component, the
+   * labels of their parent components is also shown. For `editgrid` components, the
+   * groupLabel will also be included. Using the ">" character, we separate the labels to
+   * show the relationship in the form.
+   *
+   * Example:
+   * Lets assume you have a fieldset with the label "Car", which has a soft-required
+   * textfield with the label "Brand".
+   *
+   * In this situation, the `label` will be: "Car > Brand".
+   */
+  label: string;
+}
+
+const LABEL_SEPARATOR = ' > ';
+
+export const getMissingFields = (
+  SoftRequiredComponents: AnyComponentSchema[],
+  values: object,
```

Comment:
```suggestion
  values: JSONObject,
```
?

That would also help with the type cast from earlier.